### PR TITLE
usb-cdc-ecm: resolve shellcheck issues

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

More Shellcheck fixes. The script uses a few bashisms (`echo -n`) so it is converted to a bash script.

### Testing procedure

This should be tested with the gnrc_border_router example and a board using USB CDC ECM.

### Issues/PRs references

None

### CODE REVIEW

File: dist/tools/usb-cdc-ecm/start_network.sh [hunk 1]
```diff
@@ -98,7 +98,7 @@ fi
 if [ -z "${PORT}" ]; then
     echo "Network enabled over CDC-ECM"
     echo "Press Return to stop"
-    read dummy
+    read -r
```
Comment (aabadie) [inline dist/tools/usb-cdc-ecm/start_network.sh:101]:
```suggestion
    read -r
```

**Comment by fjmolinas** [comment]
ping @bergzand !

File: dist/tools/usb-cdc-ecm/start_network.sh (hunk 1 above)
Comment (bergzand) [reply dist/tools/usb-cdc-ecm/start_network.sh]: Amended in

**Comment by bergzand** [comment]
Tested on the samr21-xpro:

<details>

```Console
koen@morgen ~/dev/RIOT-review/examples/gnrc_border_router $ make BOARD=samr21-xpro UPLINK=cdc-ecm term 
make -C ethos
make[2]: Nothing to be done for 'all'.
make -C uhcpd
make[2]: Nothing to be done for 'all'.
make -C sliptty
make[2]: Nothing to be done for 'all'.
sudo sh /home/koen/dev/RIOT-review/dist/tools/usb-cdc-ecm/start_network.sh  2001:db8::/64 /dev/ttyACM0 
Waiting for network interface.
Found interface: enp0s20f0u1u2u4
net.ipv6.conf.enp0s20f0u1u2u4.forwarding = 1
net.ipv6.conf.enp0s20f0u1u2u4.accept_ra = 0
2020-11-13 15:45:03,234 # Connect to serial port /dev/ttyACM0
Welcome to pyterm!
Type '/exit' to exit.
>
```

</details>

**Review by benpicco — APPROVED** [review]

**Review by aabadie — APPROVED** [review]
Thanks for providing the test results @bergzand ! Everything looks good.

ACK

**Pull request merged**